charts/hivemq-platform-operator/templates/deployment.yml

```diff
@@ -51,64 +51,64 @@ spec:
               value: {{.Values.javaOpts | quote }}
             {{- /* `nil` check needed for backward compatibility */ -}}
             {{- if and (ne .Values.selector "nil") (.Values.selector) }}
-            - name: quarkus.operator-sdk.controllers.hivemq-controller.selector
+            - name: QUARKUS_OPERATOR_SDK_CONTROLLERS__HIVEMQ_CONTROLLER__SELECTOR
               value: operator={{ .Values.selector }}
             {{- end }}
             - name: KUBERNETES_NAMESPACE
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
-            - name: hivemq.platform.operator.release.name
+            - name: HIVEMQ_PLATFORM_OPERATOR_RELEASE_NAME
               value: "{{ .Release.Name }}"
-            - name: hivemq.platform.operator.crd.apply
+            - name: HIVEMQ_PLATFORM_OPERATOR_CRD_APPLY
               value: "{{ .Values.crd.apply }}"
-            - name: hivemq.platform.operator.crd.wait-until-ready
+            - name: HIVEMQ_PLATFORM_OPERATOR_CRD_WAIT_UNTIL_READY
               value: "{{ .Values.crd.waitReady }}"
-            - name: hivemq.platform.operator.crd.wait-until-ready.timeout
+            - name: HIVEMQ_PLATFORM_OPERATOR_CRD_WAIT_UNTIL_READY_TIMEOUT
```

Comment:
Wasn't sure about these ones with a hyphen, but it seems it's resolved as well without any issues.
It seems that in case that both env var `FOO_BAR_BAZ=VALUE` and an application property `foo.bar-baz=default` exist, the result will be: `foo.bar-baz=value` 
https://smallrye.io/smallrye-config/Main/config/environment-variables/

Reply:
I've fixed the naming to be unambiguous regarding hyphens.